libbeat/processors/syslog/syslog_test.go

```diff
@@ -28,23 +28,21 @@ import (
 	"github.com/elastic/beats/v7/libbeat/common/cfgtype"
 )
 
-func mustParseTime(layout string, value string) time.Time {
-	t, err := time.Parse(layout, value)
+func mustParseTime(value string) time.Time {
+	t, err := time.Parse(time.RFC3339Nano, value)
 	if err != nil {
 		panic(err)
 	}
 
 	return t
 }
 
-func mustParseTimeLoc(layout string, value string, loc *time.Location) time.Time {
-	t, err := time.ParseInLocation(layout, value, loc)
+func mustParseTimeLoc(value string, loc *time.Location) time.Time {
+	t, err := time.ParseInLocation(time.Stamp, value, loc)
 	if err != nil {
 		panic(err)
 	}
-	if layout == time.Stamp {
-		t = t.AddDate(time.Now().In(loc).Year(), 0, 0)
-	}
+	t = t.AddDate(time.Now().In(loc).Year(), 0, 0)
```

Comment:
`t = t.AddDate(time.Now().In(loc).Year()-t.Year()+1, 0, 0)` or add a comment about why this is being done (the parameter provided context, but now that is lost).

Alternatively, the parameter could come back and the two functions be merged into `func mustParseTimeLoc(layout string, value string, loc *time.Location) time.Time` where the `loc` is not used for formats that provide it (e.g. called like `mustParseTime(time.RFC3339Nano, "2003-10-11T22:14:15.003Z", nil)` and `mustParseTime(time.Stamp, "Oct 11 22:14:15", cfgtype.MustNewTimezone("America/Chicago").Location())`).

Reply:
This is one of the reasons why I didn't want to remove the parameter. Context is being lost on why this is being done. What I had before wasn't perfect and only added the year when `time.Stamp` was used. Alternatively, I can look at the time's year and if it's zero I'll enrich it with the current year. I'd rather do that than blindly apply this to all timestamps that are seen (no need to do it if the time already has a valid year).

 I like what you're suggesting in the second paragraph. I never liked getting rid of the layout parameter since meaning was lost on the caller side. I'll combine the two functions into one and update the calls, and add some doc strings as well.

Reply:
This is the unfortunate consequence of including style linting in the CI failure path; it is entirely possible for a linter to push towards worse code.